imblearn/over_sampling/_rose.py

```diff
@@ -0,0 +1,121 @@
+"""Class to perform over-sampling using ROSE."""
+
+import numpy as np
+from scipy import sparse
+
+from sklearn.utils import check_random_state
+
+from .base import BaseOverSampler
+from ..utils._validation import _deprecate_positional_args
+
+
+class ROSE(BaseOverSampler):
+
+    """Oversample using Random OverSampling Examples (ROSE) algorithm.
+
+    Read more in the :ref:`User Guide <rose>`.
+    Parameters
+    ----------
+    {sampling_strategy}
+    {random_state}
+    shrink_factors : dict of {classes: shrinkfactors} couples, applied to
+        the gaussian kernels
+    {n_jobs}
+
+    Notes
+    -----
+    TODO: Support for multi-class resampling. A one-vs.one scheme is used.
+    References
+    ----------
+    .. [1] N. Lunardon, G. Menardi, N.Torelli, "ROSE: A Package for Binary
+       Imbalanced Learning," R Journal, 6(1), 2014.
+
+    .. [2] G Menardi, N. Torelli, "Training and assessing classification
+       rules with imbalanced data," Data Mining and Knowledge
+       Discovery, 28(1), pp.92-122, 2014.
```

Comment:
These are great to include. References should be referenced from the main docstring for the class. A short summary of the method would also be good.

Here's an example from the `BorderlineSMOTE` class: https://github.com/scikit-learn-contrib/imbalanced-learn/blob/0acd717578672e5d910107e81b5fb80830de54ea/imblearn/over_sampling/_smote.py#L219-L224

Reply:
I added a better description in the docstring. 

I would like to add more complete information on the maths in the docs too, but I'm not familiar with Sphinx. Could you point me to some instructions or metadocumentation?

Reply:
Math typesetting should look familiar if you've seen LaTeX before, here's a short guide from sphinx's docs: https://www.sphinx-doc.org/en/1.0/ext/math.html. Syntax is based on [reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html), which feels similar to markdown but has a powerful directive system.

> Could you point me to some instructions or metadocumentation?

Nothing specific to imblearn. If you want to learn more, sphinx's ["Getting Started"](https://www.sphinx-doc.org/en/master/usage/quickstart.html) guide is a good place to start. (I'd recommend it regardless. Sphinx is used for a huge number of projects, so the skill is extremely transferable).

Our [`Makefile`](https://github.com/scikit-learn-contrib/imbalanced-learn/blob/master/doc/Makefile) and [`conf.py`](https://github.com/scikit-learn-contrib/imbalanced-learn/blob/master/doc/conf.py) in the `docs/` directory are fairly standard. Building local documentation then looks like:

```bash
cd docs/
make html
xdg-open _build/html/index.html
```

Reply:
Ok, I learnt about Sphinx and reST. I have an issue, tho: I wanted to add a citation. I added the entry in refs.bib as

```
@article{torelli2014rose,
  title={Training and assessing classification rules with imbalanced data},
  author={Menardi G and Torelli N},
  journal={Data Mining and Knowledge Discovery},
  volume={28},
  pages={92-122},
  year={2014},
  publisher={Springer}
}
```
and it correctly shows in Bibliography page.

Then I tried 
``` 
:cite:`torelli2014rose` 
```

but the compiler says:

```bash
hecking consistency... /home/andrea/tesi/imblearn/imbalanced-learn/doc/generated/imblearn.utils.estimator_checks.parametrize_with_checks.rst: WARNING: document isn't included in any toctree
/home/andrea/tesi/imblearn/imbalanced-learn/doc/references.rst: WARNING: Citation [torelli2014rose] is not referenced.
```
and the citation is not a link in the compiled HTML.

Do I have to modify other files to obtain a link?

Reply:
Ok. I will just have some problem with plots, but I'll manage to add everything to the docs.

I have another issue: failing checks, see below. It's not very clear to me what they do address at, and what to fix.